src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -124,9 +124,17 @@ StartMetadatSyncToNode(char *nodeNameString, int32 nodePort)
 								escapedNodeName, nodePort)));
 	}
 
+	if (NodeIsCoordinator(workerNode))
+	{
+		ereport(NOTICE, (errmsg("%s:%d is the coordinator and already contains "
+								"metadata, skipping syncing the metadata",
+								nodeNameString, nodePort)));
+		return;
+	}
+
 	MarkNodeHasMetadata(nodeNameString, nodePort, true);
 
-	if (!WorkerNodeIsPrimary(workerNode))
+	if (!NodeIsPrimary(workerNode))
```

Comment:
> Do we actually need this change given that we returned for the coordinator?
> 
> I think it is easier to read the code with the change, just wanted to to make sure not to miss any details for myself.

I am not sure if I understand your point. This is not a change, we just renamed `WorkerNodeIsPrimary()` to `NodeIsPrimary()`, and it is to not sync metadata directly to secondaries since they receive the changes using streaming replication from the primary.